testament/testament.nim

```diff
@@ -1220,155 +1457,861 @@ proc parseArgs(execState: var Execution, p: var OptParser): ParseCliResult =
 
   execState.userTestOptions = p.cmdLineRest
 
-proc main() =
-  ## Define CLI Options/Args Parsing loops
-  var
-    execState = Execution(flags: {outputColour, logBackend})
-    p = initOptParser()
+if paramCount() == 0:
```

Comment:
Imperative code arbitrarily placed in the middle of a series of definitions is harder to follow, leave the main in our retractor further so it's all collected once and at the end. The former is more preferable.

Reply:
I think this one came from the rebase, don't remember writing it explicitly. I will move it out later then

Reply:
I hope I didn't do that or I need to have a conversation with myself.

Reply:
Well, not *directly*, more like a leftover https://github.com/nim-works/nimskull/pull/135/files#diff-a161d4667e86146f2f8003f08f765b8d9580ae92ec5fb6679c80c07a5310a551L1248 from the earlier times

Reply:
Thanks, I feel less bad. 😁